fdmi/plugins/motr-fdmi-app.md

```diff
@@ -1,33 +1,70 @@
+This application was written in C using `motr/client.h` interface and it adds the new Motr client to the cluster and start listening for FDMI records or events that matches the FDMI filter substrings.  
 
-The output will show whether the test was successful or not. 
+The executable will be compiled as part of the initial Motr compilation using the Makefile.am dependency and the Makefile.sub file.
 
-A second way to test this fdmi_sample_plugin will be by directly running the fdmi_sample app wrapper with the following command:
+This program then communicates with the fdmi_app python wrapper by printing to stdout the records. 
+
+To test this application we need to run the fdmi_sample app wrapper typing in the console the following command:
 
 `./fdmi_app`
 
-This method will need to have a Cortx-motr cluster already up and running beforehand that includes FDMI capabilities by adding a fdmi_filters section in the CDF.yaml config file. 
-You can check if your cluster is running with FDMI filter capability by looking at the /etc/motr/confd.xc file in which case will have the specific filter-id for the fdmi_sample_plugin. 
+The basic arguments needed will be picked by default from the `etc/motr/confd.xc` config file if not specified at the time of running.
 
-Example of such record in /etc/motr/confd.xc file will be:
+We will need to have a CORTX-Motr cluster already up and running with FDMI capabilities by adding the fdmi_filters section in the CDF.yaml config file that it is used for deploying with `hctl` command.
+ 
+You can check if your cluster is running with FDMI filter capability by looking at the /etc/motr/confd.xc file in which case will have the specific filter-id specified. 
 
-' {0x6c| ((^l|1:81), 2, ^l|1:81, "", ^n|1:3, ^v|1:66, [1: "Bucket-Name"], [1: "192.168.12.2@tcp:12345:4:1"])},'
+A second sample test that also executes the fdmi_sample_plugin program, you can be run using the following command inside this directory:
 
-More details about the FDMI design and settings can be found in this link: 
+`./fdmi_plugin_st.sh`
+
+This shell script starts Motr services using a specific filter configuration in which case we do not need to deploy the CORTX-Motr cluster beforehand using `hctl` command and specifying the fdmi_filter subsection in the CDF.yaml config file.
+
+However, we need to modify manually the `/m0t1fs/linux_kernel/st/m0t1fs_common_inc.sh` withing the `build_conf()` function to add the FDMI_FILTER_SUBSTRINGS and the FDMI_FILTER to include the number of substrings used.
 
-[FDMI Design](https://github.com/Seagate/cortx-motr/blob/main/fdmi/fdmi.c).
+The fdmi_plugin_st script starts the fdmi_plugin_sample application to listen for specific records that match with the filter already setup and do some key-value operations that reflects this specifications. 
```

Comment:
s/fdmi_plugin_st/fdmi_plugin_st.sh

Reply:
This is not fixed.
```
The `fdmi_plugin_st.sh` script starts the fdmi_plugin_sample application ...
```

Reply:
Hi @huanghua78 I just change the "The fdmi_plugin_st.sh" to be "The `fdmi_plugin_st.sh`" .
Is this sufficient?

Reply:
yes